src/client/pages/Offer/index.tsx

```diff
@@ -0,0 +1,221 @@
+import { History } from 'history'
+import { SemanticEvents } from 'quepasa'
+import React, { useEffect } from 'react'
+import {
+  useHistory,
+  useRouteMatch,
+  RouteComponentProps,
+  Redirect,
+} from 'react-router'
+import { LoadingPage } from 'components/LoadingPage'
+import { TopBar } from 'components/TopBar'
+import { Page } from 'components/utils/Page'
+import { useQuoteCartQuery, QuoteBundleVariant } from 'data/graphql'
+import { useVariation, Variation } from 'utils/hooks/useVariation'
+import { trackOfferGTM, EventName } from 'utils/tracking/gtm'
+import { getUtmParamsFromCookie, TrackAction } from 'utils/tracking/tracking'
+import { localePathPattern } from 'l10n/localePathPattern'
+import { Features, useFeature } from 'utils/hooks/useFeature'
+import { useSelectedInsuranceTypes } from 'utils/hooks/useSelectedInsuranceTypes'
+import { useCurrentLocale } from 'l10n/useCurrentLocale'
+import { LanguagePicker } from '../Embark/LanguagePicker'
+import {
+  getOfferData,
+  getUniqueQuotesFromVariantList,
+  getBundleVariantFromInsuranceTypes,
+  getInsuranceTypesFromBundleVariant,
+} from '../OfferNew/utils'
+import { AppPromotionSection } from '../OfferNew/AppPromotionSection'
+import { Checkout } from '../OfferNew/Checkout'
+import { FaqSection } from '../OfferNew/FaqSection'
+import { Introduction } from '../OfferNew/Introduction'
+import { Perils } from '../OfferNew/Perils'
+import { InsuranceSelector } from '../OfferNew/InsuranceSelector'
+
+const createToggleCheckout = (history: History<any>, locale?: string) => (
+  isOpen: boolean,
+) => {
+  if (isOpen) {
+    history.push(`/${locale}/new-member/sign`)
+  } else {
+    history.goBack()
+  }
+}
+
+type OfferPageProps = RouteComponentProps<{ id: string }>
+
+export const OfferPage = ({
+  match: {
+    params: { id: quoteCartId },
+  },
+}: OfferPageProps) => {
+  const { isoLocale, path: pathLocale } = useCurrentLocale()
+  const variation = useVariation()
+  const [isInsuranceToggleEnabled, isQuoteCartEnabled] = useFeature([
+    Features.OFFER_PAGE_INSURANCE_TOGGLE,
+    Features.QUOTE_CART_API,
+  ])
+
+  const [
+    selectedInsuranceTypes,
+    setSelectedInsuranceTypes,
+  ] = useSelectedInsuranceTypes()
```

Comment:
We need a way to pre-select a bundle. The current "bundle ID" is kind of a made-up concept which is why I don't like using it. I think it will be hard to Embark-editors to understand it. Therefore I prefer using a list of "insurance type".

Before we used a list of "quote ID" so it's not really that different, no?

99% of the time we will get `selectedInsuranceTypes` from the URL on the first page load. It being an empty array is just an edge case where we fall back to showing the first possible bunde. Or am I missing some complexity? I think the previous solution where we persisted the "quote ID" list in session storage was even harder to reason about 😅

Reply:
Got it. Thanks for the clarification.

Another thing that I've noticed is that multiple insurance types are being defined on its own `type` query paramenter. E.g: `?type=TYPE_1&type=TYPE_2`
I used to work with a [library](https://github.com/pbeshai/use-query-params/tree/master/packages/use-query-params#param-types) that supports lists as query parameters valid types. So the same example would be described like the following: `?type=TYPE_1,TYPE2`. Comma in this case is the separator, used to delimit the values of `type` query param. Do you think that's something we want to do?

Reply:
Yeah, I was considering that. Might be nicer. I just went with the standard approach since it works with `SearchParams.getAll`.